### PR TITLE
fix: add missing port when use domain

### PR DESCRIPTION
**Pull request by li-jin-gou**
#### What type of PR is this?
- fix

#### What this PR does / why we need it (English/Chinese):
- en: add missing port when use domain
- zh: 使用域名时添加默认端口

### CODE REVIEW

**Review by Duslia — DISMISSED** [review]

**Review by Duslia — APPROVED** [review]

File: pkg/app/client/client_test.go [hunk 1]
```diff
@@ -1888,3 +1889,30 @@ func TestClientReadResponseBodyStreamWithDoubleRequest(t *testing.T) {
...
+		assert.Assert(t, addr == "www.cloudwego.io:443")
+		return dialer.DialConnection(opt.Network, opt.Addr, time.Second, nil)
+	}))
+	req, res := protocol.AcquireRequest(), protocol.AcquireResponse()
+	defer func() {
+		protocol.ReleaseRequest(req)
+		protocol.ReleaseResponse(res)
+	}()
+	req.Header.SetMethod(consts.MethodGet)
+	req.SetRequestURI("https://www.cloudwego.io")
```
Comment (welkeyever) [inline pkg/app/client/client_test.go:1913]: 再加一个case：显示指定一个非443的，看看实际生效会不会被覆盖吧

Comment (li-jin-gou) [reply pkg/app/client/client_test.go:1913]: GET

Comment (li-jin-gou) [reply pkg/app/client/client_test.go]: done

**Review by welkeyever — APPROVED** [review]
lgtm

**Review by FGYFFFF — APPROVED** [review]

**Pull request merged**